tests/system/src/test/java/test/robot/javafx/scene/SRGBTest.java

```diff
@@ -0,0 +1,263 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.robot.javafx.scene;
+
+import javafx.geometry.Bounds;
+import javafx.geometry.Point2D;
+import javafx.geometry.Rectangle2D;
+import javafx.scene.Scene;
+import javafx.scene.image.PixelReader;
+import javafx.scene.image.WritableImage;
+import javafx.scene.layout.HBox;
+import javafx.scene.paint.Color;
+import javafx.scene.robot.Robot;
+import javafx.scene.shape.Rectangle;
+import javafx.stage.Screen;
+import javafx.stage.Stage;
+
+import java.awt.color.ColorSpace;
+import java.awt.GraphicsEnvironment;
+import javax.swing.SwingUtilities;
+
+import com.sun.glass.ui.Application;
+import com.sun.glass.ui.Window;
+import com.sun.javafx.PlatformUtil;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.Test;
+import static org.junit.Assume.assumeTrue;
+
+import test.robot.testharness.VisualTestBase;
+
+public class SRGBTest extends VisualTestBase {
+
+    private static final int SWATCH_SIZE = 200;
+
+    // Avoid testing colors out near the edge of the gamut to avoid clipping
+    // when converting between spaces.
+    static final float LOW = 0.25f;
+    static final float MID = 0.50f;
+    static final float HIGH = 0.75f;
+
+    // The component tolerance allows one bit of rounding when writing a color
+    // out and another bit when reading it back in.
+    static final double COMPONENT_TOLERANCE = 2.0 / 255.0;
+
+    private enum TestColor {
+        COLOR_01(LOW, MID, HIGH),
+        COLOR_02(LOW, HIGH, MID),
+        COLOR_03(MID, LOW, HIGH),
+        COLOR_04(MID, HIGH, LOW),
+        COLOR_05(HIGH, LOW, MID),
+        COLOR_06(HIGH, MID, LOW);
+
+        public final float red;
+        public final float green;
+        public final float blue;
+
+        TestColor(float r, float g, float b) {
+            red = r;
+            green = g;
+            blue = b;
+        }
+    };
+
+    // We center windows in the visual bounds of the screen to make it easy
+    // for both JavaFX and AWT to sample from the same point.
+    private Point2D getJFXScreenCenter() {
+        Rectangle2D screenBounds = Screen.getPrimary().getVisualBounds();
+        double centerX = screenBounds.getMinX() + screenBounds.getWidth() / 2.0;
+        double centerY = screenBounds.getMinY() + screenBounds.getHeight() / 2.0;
+        return new Point2D(centerX, centerY);
+    }
+
+    // We use an AWT Robot since it is color space aware and will correctly convert
+    // from the screeen's color space to sRGB.
```

Comment:
do we plan to fix FX robot eventually?

Reply:
This PR fixes the JavaFX Robot. I give a more detailed explanation of why I'm using the AWT Robot in a comment just before the sRGBPixelTest. I've updated the comments.